src/awkward/operations/ak_almost_equal.py

```diff
@@ -82,6 +105,10 @@ def packed_list_content(layout):
         return layout.content[layout.offsets[0] : layout.offsets[-1]]
 
     def visitor(left, right) -> bool:
+        # Most firstly, check same_content_types before any transformations
+        if same_content_types and left.__class__ != right.__class__:
```

Comment:
```suggestion
        if same_content_types and left.__class__ is right.__class__:
```

I'm surprised that a Python linter didn't complain about that, but class objects should be checked for reference equality, since they ought to be singletons (like `None`). In fact, a linter might even complain about the mere fact that you're checking class equality, rather than `issubclass(left.__class__, right.__class__)`, though you really do want class equality, not inheritance.